extras/menus/advancedvsh/advanced.c

```diff
@@ -46,385 +46,416 @@ extern char* available_fonts[];
+				blit_set_color(fc,bc);
+				break;
+			// Light Green
+			case 8:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa000b300;
+				blit_set_color(fc,bc);
+				break;
+			// Blue
+			case 9:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00ff0000;
+				blit_set_color(fc,bc);
+				break;
+			// Light Blue
+			case 10:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0ff0000;
+				blit_set_color(fc,bc);
+				break;
+			// Indigo
+			case 11:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x0082004b;
+				blit_set_color(fc,bc);
+				break;
+			// Light Indigo
+			case 12:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa082004b;
+				blit_set_color(fc,bc);
+				break;
+			// Violet
+			case 13:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00ee82ee;
+				blit_set_color(fc,bc);
+				break;
+			// Light Violet
+			case 14:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0ee82ee;
+				blit_set_color(fc,bc);
+				break;
+			// Pink 
+			case 15:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00cbc0ff;
+				blit_set_color(fc,bc);
+				break;
+			// Light Pink 
+			case 16:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0cbc0ff;
+				blit_set_color(fc,bc);
+				break;
+			// Purple 
+			case 17:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00993366;
+				blit_set_color(fc,bc);
+				break;
+			// Light Purple 
+			case 18:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0993366;
+				blit_set_color(fc,bc);
+				break;
+			// Teal 
+			case 19:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00808000;
+				blit_set_color(fc,bc);
+				break;
+			// Light Teal 
+			case 20:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0808000;
+				blit_set_color(fc,bc);
+				break;
+			// Aqua 
+			case 21:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00cccc00;
+				blit_set_color(fc,bc);
+				break;
+			// Light Aqua 
+			case 22:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0cccc00;
+				blit_set_color(fc,bc);
+				break;
+			// Grey 
+			case 23:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00737373;
+				blit_set_color(fc,bc);
+				break;
+			// Light Grey 
+			case 24:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0737373;
+				blit_set_color(fc,bc);
+				break;
+			// Black 
+			case 25:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00000000;
+				blit_set_color(fc,bc);
+				break;
+			// Light Black 
+			case 26:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xa0000000;
+				blit_set_color(fc,bc);
+				break;
+			// White  
+			case 27:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0x00ffffff;
+				blit_set_color(fc,bc);
+				break;
+			// Light White  
+			case 28:
+				bc = (submax_menu==submenu_sel) ? 0x0000ff : 0xafffffff;
+				blit_set_color(fc,bc);
+				break;
+			default:	
+				bc = (submax_menu==submenu_sel) ? 0xff8080 : 0x0000a5ff;
+				blit_set_color(fc,bc);
+		}
 
+		// do fg_color stuff
+		switch(config.vsh_fg_color) {
+			// Random
+			case 0:
```

Comment:
Looks like this was an issue before, but 0 would fall through to then be case 1, should have a break possibly.